PubNub/Data/Models/PNPAMToken.h

```diff
@@ -0,0 +1,161 @@
+};
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+#pragma mark - Interfaces declaration
+
+/**
+ * @brief \b PubNub Access Manager resource permission.
+ *
+ * @discussion Object represent single resource permission information.
+ *
+ * @author Serhii Mamontov
+ * @version 4.17.0
+ * @since 4.17.0
+ * @copyright © 2010-2021 PubNub, Inc.
+ */
+@interface PNPAMResourcePermission : NSObject
+
+
+#pragma mark - Information
+
+/**
+  * @brief Value with which instance has been initialized.
+ */
+@property (nonatomic, readonly, assign) PNPAMPermission value;
+
+
+#pragma mark -
+
+
+@end
+
+
+/**
+ * @brief \b PubNub Access Manager resource permissions.
+ *
+ * @discussion Object represent resource permissions stored in PAM token.
+ *
+ * @author Serhii Mamontov
+ * @version 4.17.0
+ * @since 4.17.0
+ * @copyright © 2010-2021 PubNub, Inc.
+ */
+@interface PNPAMTokenResource : NSObject
+
+
+#pragma mark - Information
+
+/**
+ * @brief Permissions granted to specific / regexp matching channels.
+ */
+@property (nonatomic, readonly, strong) NSDictionary<NSString *, PNPAMResourcePermission *> *channels;
+
+/**
+ * @brief Permissions granted to specific / regexp matching channel groups.
+ */
+@property (nonatomic, readonly, strong) NSDictionary<NSString *, PNPAMResourcePermission *> *groups;
+
+/**
+ * @brief Permissions granted to specific / regexp matching uuids.
+ */
+@property (nonatomic, readonly, strong) NSDictionary<NSString *, PNPAMResourcePermission *> *uuids;
+
+#pragma mark -
+
+
+@end
+
+
+/**
+ * @brief \b PubNub Access Manager Token.
+ *
+ * @discussion Object represent decoded token content.
+ *
+ * @author Serhii Mamontov
+ * @version 4.17.0
+ * @since 4.17.0
+ * @copyright © 2010-2021 PubNub, Inc.
+ */
+@interface PNPAMToken : NSObject
+
+
+#pragma mark - Information
+
+/**
+ * @brief Token version.
+ */
+@property (nonatomic, readonly, assign) NSUInteger version;
+
+/**
+ * @brief Token generation date time.
+ */
+@property (nonatomic, readonly, assign) NSUInteger timestamp;
+
+/**
+ * @brief Maximum duration (in minutes) during which token will be valid.
+ */
+@property (nonatomic, readonly, assign) NSUInteger ttl;
+
+/**
+ * @brief The uuid that is exclusively authorized to use this token to make API requests.
+ */
+@property (nonatomic, nullable, readonly, strong) NSString *authorizedUUID;
+
+/**
+ * @brief Permissions granted to specific resources.
+ */
+@property (nonatomic, readonly, strong) PNPAMTokenResource *resources;
+
+/**
+ * @brief Permissions granted to resources which match specified regular expression.
+ */
+@property (nonatomic, readonly, strong) PNPAMTokenResource *patterns;
+
+/**
+ * @brief Additional information which has been added to the token.
+ */
+@property (nonatomic, readonly, strong) NSDictionary *meta;
+
+/**
+ * @brief PAM token content signature.
+ */
+@property (nonatomic, readonly, strong) NSData *signature;
```

Comment:
Is that only for debugging?

Reply:
@bartk what exactly, presence of `signature`? It was encoded in token, so I provide it as well.

Reply:
Ok, I just notice that we have some lack of consistency here among SDK. For me either way is ok. Maybe we shall refer to @raipubnub to confirm if this would be required also in other SDKs....

Reply:
@bartk `signature` actually in specs, but in fact it is non-utf8, so it is just array of bytes, not string.